### ISSUE TITLE
Add graphmap2 to BioConda

### ISSUE THREAD

**Issue by drpatelh** [nf-core/nanoseq#25]
I cant seem to find `GraphMap2` on BioConda which also means the associated BioContainer wont exist. It would be nice to be able to use this in the pipeline because it has some updated functionality for Nanopore protocols.

**Comment by apeltzer** [comment]
Copying in from Slack discussion on this:

....

https://github.com/lbcb-sci/graphmap2
Which is version 0.3.0
and packaged in bioconda as graphmap https://bioconda.github.io/recipes/graphmap/README.html

a bit odd since there aren't many releases and the bioconda packaged one is from the author Ivan Sovic's repository that instead links to the new one in his (former?) group...

Also odd that the new repository doesn't have the > 0.3X releases on it - so probably should ping the authors and then make a new release with the new URL? Could also ask over at Bioconda gitter with respect to this?

**Comment by drpatelh** [comment]
Requested by @dpryan79 here https://github.com/lbcb-sci/graphmap2/issues/6

**Comment by lwratten** [comment]
The BioConda `GraphMap` version has been updated to the latest release
https://bioconda.github.io/recipes/graphmap/README.html

**Comment by apeltzer** [comment]
ok, then adding in `- bioconda::graphmap2=0.6.3` to the `environment.yml` and you should be fine 👍

**Comment by drpatelh** [comment]
Except we aren't using `conda` because `guppy` isn't available there and could be problematic adding the `gpu` version :sweat_smile: .Using a mix of containers on DockerHub and BioContainers to overcome this.

**Comment by dpryan79** [comment]
Can guppy be redistributed?

**Comment by drpatelh** [comment]
Hmm...good question :+1: Would have to ask Nanopore. In theory, it should support both `gpu` and `cpu` out of the box but in practice I couldn't use the `cpu` container for `gpu`. Possibly some `gpu` libraries missing in the container. When I looked it was quite difficult getting information regarding the software and any docs - needed to have a Nanopore account.

**Comment by dpryan79** [comment]
I have a nanopore account, so I can ask them if you'd like (they're not terribly speedy at replying).

**Comment by drpatelh** [comment]
Docker containers we are currently using in the pipeline:

GPU Docker container:
https://hub.docker.com/r/nanozoo/guppy_gpu/dockerfile

CPU Docker container:
https://hub.docker.com/r/genomicpariscentre/guppy/dockerfile

**Comment by drpatelh** [comment]
Thanks @dpryan79 . That would be amazing! I'm actually supposed to be on paternity leave. Apparently 🤪

**Comment by apeltzer** [comment]
Ok, if I'm not mistaken there, one could simply work on adding in the required CUDA libraries to the container itself: `nvidia/cuda:9.0-base`

If you're running with Singularity, these are then used to accessing the GPU(s) automatically for you.

**Comment by dpryan79** [comment]
@drpatelh I'll post a query on their forum then. Don't spend too much time on github :)

**Comment by dpryan79** [comment]
For those with an ONT account I've asked about guppy redistribution: https://community.nanoporetech.com/posts/guppy-redistribution

**Comment by dpryan79** [comment]
So, the reply from ONT is that guppy cannot be redistributed at the moment. The docker containers are actually breaches of their license, which I suppose isn't an issue for nf-core (you're just using them, not providing them), but the people who made them could theoretically get sued. Granted, the odds of that are very low (especially since ONT is allegedly going to at least provide their own containers somewhere), but they're not 0. BTW, for those of us with clusters, we're actually supposed to restrict access to guppy to only ONT customers...not sure how that's really possible in practice without them having their own copies.

**Comment by drpatelh** [comment]
Thanks @dpryan79. So basically it can't be added to BioConda (unless the license is changed or we get explicit approval). Not sure what to say about the people hosting the containers on Dockerhub - I think they are actually doing the Bioinformatics community a service by hosting them before they are officially added but I guess it doesn't matter what I think. As far as I know, only people with an official account are using the pipeline at the Crick but this is difficult to guage once they are made available to download and run.

**Comment by dpryan79** [comment]
Yeah, ONT needs to just relax their licensing and make everyone's life easier.

**Comment by apeltzer** [comment]
Maybe something to consider as a successor to Guppy, apparently even better: https://github.com/nf-core/nanoseq/issues/61
